tests/unit/tests_high_five_base.cpp

```diff
@@ -1531,6 +1532,83 @@ BOOST_AUTO_TEST_CASE(HighFiveFixedLenStringArrayStructure) {
     }
 }
 
+BOOST_AUTO_TEST_CASE(HighFiveReference) {
+    const std::string FILE_NAME("h5_ref_test.h5");
+    const std::string DATASET1_NAME("dset1");
+    const std::string DATASET2_NAME("dset2");
+    const std::string GROUP_NAME("/group1");
+    const std::string REFGROUP_NAME("/group2");
+    const std::string REFDATASET_NAME("dset2");
+
+    ContentGenerate<double> generator;
+    std::vector<double> vec1(4);
+    std::vector<double> vec2(4);
+    std::generate(vec1.begin(), vec1.end(), generator);
+    std::generate(vec2.begin(), vec2.end(), generator);
+    {
+        // Create a new file using the default property lists.
+        File file(FILE_NAME, File::ReadWrite | File::Create | File::Truncate);
+
+        // create group
+        Group g1 = file.createGroup(GROUP_NAME);
+
+        // Create the data space for the dataset.
+        std::vector<size_t> dims{4};
+        DataSpace dataspace(dims);
+
+        DataSet dataset1 = g1.createDataSet(
+            DATASET1_NAME,
+            dataspace, AtomicType<double>());
+        DataSet dataset2 = g1.createDataSet(
+            DATASET2_NAME,
+            dataspace, AtomicType<double>());
+
+        // write some data
+        dataset1.write(vec1);
+        dataset2.write(vec2);
+
+        // create group and dataset to hold reference
+        Group refgroup = file.createGroup(REFGROUP_NAME);
+
+        // Create the data space for the dataset.
+        std::vector<size_t> ref_dims{2};
+        DataSpace ref_dspace(ref_dims);
+
+        DataSet ref_ds = refgroup.createDataSet(REFDATASET_NAME, ref_dspace, AtomicType<Reference>());
+
+        auto references = std::vector<Reference>({{g1, dataset1},{file, g1}});
+
+        ref_ds.write(references);
```

Comment:
Actually, we should be able to use the shortcut syntax. Add a test to prove so please.
`refgroup.createDataSet(REFDATASET_NAME, references)`